src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -524,30 +530,38 @@ public function getNamespaces()
      */
     protected function getNamespacesObject()
     {
-        if ($this->namespaces->count() === 0) {
-            $query = 'SELECT * FROM phpcr_namespaces';
-            $data = $this->getConnection()->fetchAll($query);
-
-            $this->namespaces->exchangeArray(array(
+        if (null === $this->namespaces) {
+            $namespaces = array(
                 NamespaceRegistryInterface::PREFIX_EMPTY => NamespaceRegistryInterface::NAMESPACE_EMPTY,
                 NamespaceRegistryInterface::PREFIX_JCR => NamespaceRegistryInterface::NAMESPACE_JCR,
                 NamespaceRegistryInterface::PREFIX_NT => NamespaceRegistryInterface::NAMESPACE_NT,
                 NamespaceRegistryInterface::PREFIX_MIX => NamespaceRegistryInterface::NAMESPACE_MIX,
                 NamespaceRegistryInterface::PREFIX_XML => NamespaceRegistryInterface::NAMESPACE_XML,
                 NamespaceRegistryInterface::PREFIX_SV => NamespaceRegistryInterface::NAMESPACE_SV,
-            ));
+            );
 
-            foreach ($data as $row) {
-                $this->namespaces[$row['prefix']] = $row['uri'];
-            }
+            $query = 'SELECT prefix, uri FROM phpcr_namespaces';
+            $result = $this->getConnection()->query($query);
+            $namespaces += $result->fetchAll(\PDO::FETCH_KEY_PAIR);
 
-            $this->originalNamespaces = $this->namespaces;
+            $this->setNamespaces($namespaces);
         }
 
         return $this->namespaces;
     }
 
     /**
+     * Set the namespaces property while also storing a backup into originalNamespaces
+     *
+     * @param array $namespaces
+     */
+    protected function setNamespaces(array $namespaces)
+    {
+        $this->namespaces = new \ArrayObject($namespaces);
```

Comment:
this was taken care of already, since I was using this method in case of a rollback and not this setter. but I have rewritten things to be more clear.